### DISCUSSION TITLE
Why is there a UDP 0.0.0.0 listening every time a DnsNameResolver is initialized?

### DISCUSSION THREAD

**Discussion by llchry** [netty/netty#14013] [Q&A] [answered]
```
        EventLoopGroup group = new NioEventLoopGroup();
        DnsNameResolver resolver = new DnsNameResolverBuilder(group.next())
                .channelType(NioDatagramChannel.class)
                .maxQueriesPerResolve(1)
                .queryTimeoutMillis(1000)
                .resolvedAddressTypes(ResolvedAddressTypes.IPV6_PREFERRED)
                .nameServerProvider(new SingletonDnsServerAddressStreamProvider(new InetSocketAddress("8.8.8.8",53)))
                .build();
        Future<InetAddress> future = resolver.resolve("www.google.com");
        InetAddress address = future.get();
        System.out.println(address);
```

  UDP    0.0.0.0:58163          *:*                                    23340
  UDP    0.0.0.0:58164          *:*                                    23340
  UDP    [::]:58163             *:*                                    23340
  UDP    [::]:58164             *:*                                    23340

**Comment by chrisvest** [comment]
The socket needs to be bound to an address in order to receive replies to your queries.

**Reply by llchry** [reply]
Why do you have to start a monitor? I don't understand. What good is it? If I don't want to have such a listener, is there a configuration to turn it off?

**Reply by normanmaurer** [reply]
How would it receive a reply if it would not bind to an address ? I am not sure what you are asking for.

**Reply by llchry** [reply]
Oh, I see. I thought this was meant to be a server port.  A typical request to the server is a connection like this, but here the request is very different :
`tcp6 0 0 192.168.24.245:26328 192.168.24.36:57382 ESTABLISHED 4393/java`

**Reply by normanmaurer** [reply, marked as answer]
it uses UDP which is a connection-less protocol
